### PR TITLE
Add Glob / Wildcard support to Cloudwatch input for 'Dimensions' configuration

### LINKED ISSUE #4046: Input: CloudWatch - Partial Wildcard or Regexp support?
## Feature Request

Opening a feature request kicks off a discussion.

### Proposal: 

Support Partial Wildcard (e.g. `*prod*`, like glob) or RegExp support (e.g. `/prod/`) to filter out unwanted metrics

For example, below configuration will pull production metrics only:

Using RegExp:

```toml
[[inputs.cloudwatch]]
  namespace = "AWS/Lambda"
  [[inputs.cloudwatch.metrics]]
    names = ["Invocations", "Errors", "Duration"]
    [[inputs.cloudwatch.metrics.dimensions]]
      name = "FunctionName"
      value = "/prod/"
```

... or partial wildcard (globbing): 

```toml
[[inputs.cloudwatch]]
  namespace = "AWS/Lambda"
  [[inputs.cloudwatch.metrics]]
    names = ["Invocations", "Errors", "Duration"]
    [[inputs.cloudwatch.metrics.dimensions]]
      name = "FunctionName"
      value = "*prod*"
```

### Current behavior:

There's no easy way to select production-only metrics without specifying them.

### Desired behavior:

Users can filter metrics using Glob or RegExp

### Use case:

Since our architecture of our application is microservices, We have 300+ lambda functions and we have 3 stages (production, staging, development) and number of functions are growing.
We replicate cloudwatch metrics to our InfluxDB using Telegraf with Cloudwatch Input Plugin to detect anomaly metrics (e.g. Latency Spikes). When something goes wrong, we receive alarms from Bosun.

Anyway currently filtering production resources is so hard. everytime we release new "service" to  production, we need to change bunch line in the telegraf configuration to filter non-production metrics.

### PR DESCRIPTION
**Pull request by Jagularr**
I believe this will resolve #4046

### CODE REVIEW

File: plugins/inputs/cloudwatch/README.md [hunk 1]
```diff
@@ -101,6 +101,7 @@ API endpoint. In the following order the plugin will attempt to authenticate.
   #
   #  ## Dimension filters for Metric.  All dimensions defined for the metric names
   #  ## must be specified in order to retrieve the metric statistics.
+  #  ## 'value' has wildcard / 'glob' matching support.
```
Comment (srebhan) [inline plugins/inputs/cloudwatch/README.md:104]: Could you please add a small example like
```suggestion
  #  ## 'value' has wildcard / 'glob' matching support such as `p-*`.
```
Furthermore, you should also add this to the `sampleConfig`...

File: plugins/inputs/cloudwatch/cloudwatch_test.go [hunk 2]
```diff
@@ -201,17 +201,19 @@ func TestSelectMetrics(t *testing.T) {
...
-						Value: "*",
+						Value: "us-east*",
 					},
 				},
 			},
 		},
 	}
+	err := c.initializeCloudWatch()
 	c.client = &mockSelectMetricsCloudWatchClient{}
+	assert.NoError(t, err)
```
Comment (srebhan) [inline plugins/inputs/cloudwatch/cloudwatch_test.go:216]: Can you please check the error _directly_ after setting `err`, i.e. move this one line up?!

**Review by srebhan — COMMENTED** [review]
Seems reasonable to me. Some small comments in the code.

**Comment by Jagularr** [comment]
!retry-checks

**Review by srebhan — APPROVED** [review]
Looks good to me.

**Review by sspaink — APPROVED** [review]

**Pull request merged**